src/libappimage/core/FilesIterator.cpp

```diff
@@ -0,0 +1,76 @@
+// local
+#include <appimage/core/FilesIterator.h>
+#include "core/impl/TraversalFallback.h"
+#include "core/impl/TraversalType1.h"
+#include "core/impl/TraversalType2.h"
+
+using namespace appimage::core;
+
+FilesIterator::FilesIterator(std::string path, FORMAT format) : last(new impl::TraversalFallback()) {
+    switch (format) {
+        case TYPE_1:
+            priv = std::shared_ptr<Traversal>(new impl::TraversalType1(path));
```

Comment:
We should try to avoid any form of (slow) virtual binding for such low-level stuff by using templating.

Reply:
I guess that I'll have to refresh my studies on C++ Templates.
Creating an issue for it #52